includes/payment-methods/class-upe-payment-method.php

```diff
@@ -252,7 +252,9 @@ public function is_currency_valid( string $account_domestic_currency, $order_id
 			}
 		}
 
-		return empty( $this->currencies ) || in_array( $current_store_currency, $this->currencies, true );
+		$supported_currencies = $this->get_currencies();
+
+		return empty( $supported_currencies ) || in_array( $current_store_currency, $supported_currencies, true );
```

Comment:
I added this modification just so that we consistently rely on the `get_currency()` getter, rather than directly relying on the `currencies` property.